packages/components/src/higher-order/with-filters/index.js

```diff
@@ -24,45 +24,82 @@ const ANIMATION_FRAME_PERIOD = 16;
  */
 export default function withFilters( hookName ) {
 	return createHigherOrderComponent( ( OriginalComponent ) => {
-		return class FilteredComponent extends Component {
-			/** @inheritdoc */
+		const namespace = 'core/with-filters/' + hookName;
+
+		/**
+		 * Since filtering is applied to the component, each filtered instance
+		 * can reuse a shared reference to the definition. This optimizes to
+		 * avoid excessive calls to `applyFilters` when many instances exist.
+		 *
+		 * @type {Component}
+		 */
+		let FilteredComponent = applyFilters( hookName, OriginalComponent );
```

Comment:
I suspect this line might be the reason for failing e2e tests.

Before, we would call it in the constructor, which is much later in the flow - as it is after the editor starts to render. In this case, it happens just after HOCs are applied - which happens just after the component is exported from the JS module.

Reply:
> I suspect this line might be the reason for failing e2e tests.
> 
> Before, we would call it in the constructor, which is much later in the flow - as it is after the editor starts to render. In this case, it happens just after HOCs are applied - which happens just after the component is exported from the JS module.

That makes sense. Maybe then we can have this be returned from some getter function which computes the value once when it's called.

Like Lodash's `once`:

https://lodash.com/docs/4.17.11#once